pennylane/workflow/_capture_qnode.py

```diff
@@ -298,6 +300,8 @@ def _qnode_jvp(args, tangents, **impl_kwargs):
 
     batching.primitive_batchers[qnode_prim] = _qnode_batching_rule
 
+    mlir.register_lowering(qnode_prim, mlir.lower_fun(qnode_impl, multiple_results=True))
```

Comment:
What is the default lowering rule, and why did that not work in our case?

Reply:
The default lowering rule is error.

Reply:
Why does the lowering not fail for other primitives in that case?

Reply:
The other primitives aren't seen by `jax.jit`.  `jax.jit` only sees the execution of the jaxpr, not the capturing of it.

Reply:
We may also need to do this to the gradient primitives though.

Reply:
Thanks. Had a chat offline. I thought this would also impact other higher-order primitives, but `DefaultQubitInterpreter` unrolls those primitives, so we don't execute the jaxpr of other HOPs, which is why we don't need to do the same registration for them.

Approving.